### PR TITLE
Add support for auditmatica

### PR DESCRIPTION
**Pull request by scollazo**
https://github.com/artefactual-labs/auditmatica

This PR adds a new variable, archivematica_src_audit, that enables
the feature and deploys the corresponding nginx config file.

### CODE REVIEW

**Review by mamedin — APPROVED** [review]

File: templates/etc/nginx/conf.d/auditmatica.conf.j2 [hunk 1]
```diff
@@ -1,9 +1,46 @@
...
+    ~/ingest/status/(.*) 0;
+    # DataTable column state management.
+    ~/archival-storage/load_state/ 0;
+    ~/archival-storage/save_state/ 0;
+    ~/backlog/load_state/ 0;
+    ~/backlog/save_state/ 0;
+    # Web assets.
+    ~/media/ 0;
+    ~/static/ 0;
+    ~/jsi18n/ 0;
```
Comment (tw4l) [inline templates/etc/nginx/conf.d/auditmatica.conf.j2:18]: These likely need to terminate in `(.*)` as well - `auditmatica` looks for whether a URL begins with these values, but the full URL is longer and variable depending on the file being served

File: templates/etc/nginx/conf.d/auditmatica.conf.j2 [hunk 2]
```diff
@@ -1,9 +1,46 @@
...
+    # cannot access the content from here, we can safely ignore.
+    ~/filesystem/children/location 0;
+    # API calls used internally by Archivematica.
+    ~/api/ingest/copy_metadata_files 0;
+    ~/api/administration/dips/atom/levels 0;
+    ~/api/administration/dips/atom/fetch_levels 0;
+    ~/api/v2beta/jobs 0;
+    ~/api/v2beta/task 0;
+    ~/api/filesystem/metadata 0;
+    ~/api/processing-configuration 0;
```
Comment (tw4l) [inline templates/etc/nginx/conf.d/auditmatica.conf.j2:32]: This covers both `/api/processing-configuration` and `/api/processing-configuration/<name>/`, so maybe it should be `~/api/processing-configuration(.*)`

File: templates/etc/nginx/conf.d/auditmatica.conf.j2 [hunk 3]
```diff
@@ -1,9 +1,46 @@
...
+    # Calls issued while user browses transfer browser. Since users
+    # cannot access the content from here, we can safely ignore.
+    ~/filesystem/children/location 0;
+    # API calls used internally by Archivematica.
+    ~/api/ingest/copy_metadata_files 0;
+    ~/api/administration/dips/atom/levels 0;
+    ~/api/administration/dips/atom/fetch_levels 0;
+    ~/api/v2beta/jobs 0;
+    ~/api/v2beta/task 0;
+    ~/api/filesystem/metadata 0;
```
Comment (tw4l) [inline templates/etc/nginx/conf.d/auditmatica.conf.j2:31]: I believe these all may or may not end in a trailing slash

File: templates/etc/nginx/conf.d/auditmatica.conf.j2 [hunk 4]
```diff
@@ -1,9 +1,46 @@
...
+    ~/api/processing-configuration 0;
+    ~/api/v2beta/validate 0;
+    # The AJAX call issued to populate the Packages tab datatable
+    # can be ignored because the associated user activity (viewing
+    # package information) is already covered.
+    ~/packages_ajax 0;
```
Comment (tw4l) [inline templates/etc/nginx/conf.d/auditmatica.conf.j2:37]: These may also end in a trailing slash or not

File: templates/etc/nginx/conf.d/auditmatica.conf.j2 [hunk 5]
```diff
@@ -1,9 +1,46 @@
...
+}
+
+map $time_iso8601 $logdate {
+  '~^(?<ymd>\d{4}-\d{2}-\d{2})'   $ymd;
+   default                         'nodate';
 }
 
 log_format audit '$remote_addr - $remote_user [$time_local] "$request" $status $body_bytes_sent "$http_referer" "$http_user_agent" "$http_x_forwarded_for" user=$upstream_http_x_username';
-access_log "/var/log/nginx/audit.log" audit if=$loggable;
+access_log "/var/log/nginx/audit-${logdate}.log" audit if=$loggable;
```
Comment (tw4l) [inline templates/etc/nginx/conf.d/auditmatica.conf.j2:46]: Will this write to a new file every time it's a new day, or does it continue writing to a file that's named after the day it was initially created?

**Review by tw4l — COMMENTED** [review]
Hi @scollazo - overall this is looking good! I like the conditional logging - very cool. I pointed out a few places where the URLs in the `$loggable` map might need some tweaks - namely, because auditmatica checks requests to see if any begin with values in the `URLS_TO_IGNORE` list you've copied these from, but many of them have variable endings. For example, optional trailing slashes, UUIDs, or in at least one case of an API URL, multiple endpoints that share a base URL.

I left a question too about including the date in the log's filename. I'm happy to defer to your wisdom on how to deploy but I'm curious about the rationale for your thinking there.

File: templates/etc/nginx/conf.d/auditmatica.conf.j2 [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    ~/tasks/ 0;
+    # Calls issued while user browses transfer browser. Since users
+    # cannot access the content from here, we can safely ignore.
+    ~/filesystem/children/location 0;
+    # API calls used internally by Archivematica.
+    ~/api/ingest/copy_metadata_files 0;
+    ~/api/administration/dips/atom/levels 0;
+    ~/api/administration/dips/atom/fetch_levels 0;
+    ~/api/v2beta/jobs 0;
+    ~/api/v2beta/task 0;
```
Comment (tw4l) [inline templates/etc/nginx/conf.d/auditmatica.conf.j2:30]: These two specifically also typically continue with a UUID, so might be good candidates for e.g. `~/api/v2beta/jobs/(.*)`

**Comment by scollazo** [comment]
@tw4l , as far as I know, the (.*) are mostly used for rewrites, in order to create backreferences to the regular expresion. As that's not needed here, I removed them.

**Pull request merged**